src/ui_fsmenu/crash_report.cc

```diff
@@ -0,0 +1,199 @@
+	                "crash_details",
+	                0,
+	                0,
+	                0,
+	                0,
+	                _("Crash Report Details")) {
+		UI::Box* box = new UI::Box(
+		   this, UI::PanelStyle::kFsMenu, "main_box", 0, 0, UI::Box::Vertical, 0, 0, kPadding);
+
+		std::string detailed_text_rt = richtext_escape(report);
+		newlines_to_richtext(detailed_text_rt);
+		detailed_text_rt =
+		   as_richtext_paragraph(detailed_text_rt, UI::FontStyle::kFsMenuInfoPanelParagraph);
+
+		UI::MultilineTextarea* details = new UI::MultilineTextarea(
+		   box, "details", 0, 0, 700, 400, UI::PanelStyle::kFsMenu, detailed_text_rt);
+
+		UI::Box* buttons_box = new UI::Box(
+		   box, UI::PanelStyle::kFsMenu, "buttons_box", 0, 0, UI::Box::Horizontal, 0, 0, kPadding);
+
+		UI::Button* b_copy = new UI::Button(buttons_box, "copy", 0, 0, kButtonSize, 0,
+		                                    UI::ButtonStyle::kFsMenuSecondary, _("Copy report"),
+		                                    _("Copy the full report to the clipboard"));
+		UI::Button* b_close = new UI::Button(
+		   buttons_box, "close", 0, 0, kButtonSize, 0, UI::ButtonStyle::kFsMenuPrimary, _("Close"));
+
+		b_copy->sigclicked.connect([report]() { SDL_SetClipboardText(report.c_str()); });
+		b_close->sigclicked.connect([this]() { die(); });
+
+		buttons_box->add(b_copy, UI::Box::Resizing::kExpandBoth);
+		buttons_box->add(b_close, UI::Box::Resizing::kExpandBoth);
+		box->add(details, UI::Box::Resizing::kExpandBoth);
+		box->add(buttons_box, UI::Box::Resizing::kExpandBoth);
+
+		set_center_panel(box);
+		center_to_parent();
+		initialization_complete();
+	}
+};
+
+CrashReportWindow::CrashReportWindow(FsMenu::MainMenu& menu, const FilenameSet& crash_files)
+   : UI::Window(&menu, UI::WindowStyle::kFsMenu, "crash", 0, 0, 0, 0, _("Crash Report")),
+     clear_on_close_(crash_files) {
+
+	UI::Box* box = new UI::Box(
+	   this, UI::PanelStyle::kFsMenu, "main_box", 0, 0, UI::Box::Vertical, 0, 0, kPadding);
+
+	// Only for the report. TODO(Nordfriese): Just pull the report assembly out into a new function.
+	TechInfoBox tech_info(box, TechInfoBox::Type::kAbout);
+	tech_info.set_visible(false);
+
+	// not localized
+	std::string detailed_text =
+	   format("# Autogenerated crash report for %u crash(es):\n\n", crash_files.size());
+	for (const std::string& filename : crash_files) {
+		detailed_text += "\n- ";
+		detailed_text += FileSystem::fs_filename(filename.c_str());
+	}
+	detailed_text += "\n\n## Technical report:\n";
+	detailed_text += tech_info.get_report();
+	for (const std::string& filename : crash_files) {
+		detailed_text += "\n\n## ";
+		detailed_text += filename;
+		detailed_text += "\n\n";
+
+		FileRead fr;
+		try {
+			fr.open(*g_fs, filename);
+			detailed_text += "```\n";
+			detailed_text += fr.data(fr.get_size());
+			detailed_text += "```\n";
+		} catch (const std::exception& e) {
+			detailed_text += "Error opening crash report file:\n";
```

Comment:
Nice touch making it valid markdown. :)

I wonder however whether it should have a `detailed_text += "```\n";` as first step in `catch()` to close the code block in case of read errors.